File: Thirty-Day LeetCoding Challenge/Week 2/Backspace String Compare/sol.cpp
class Solution {
public:
    string func(string s){
        int len = s.size();
        stack<char> stk;
        for(int i = 0; i < len; ++i){
            char cur = s[i];
            if(cur == '#'){
                if(!stk.empty())
                    stk.pop();
            }else
                stk.push(cur);
        }
        string ret = "";
        while(!stk.empty()){
            char c = stk.top();
            stk.pop();
            ret = c + ret;
        }
        return ret;
    }
    bool backspaceCompare(string S, string T) {
        return func(S) == func(T);
    }
};

File: 0007.ReverseInteger/sol.cs
public class Solution {
    public int Reverse(int x) {
        int ret = 0;
        while(x != 0){
            int pop = x % 10;
            if(x >= 0){
                if(ret > 214748364 || (ret == 214748364 && pop >= 7))
                    return 0;
                ret = ret * 10 + pop;
            }else{
                if(ret < -214748364 || (ret == -214748364 && pop < -8))
                    return 0;
                ret = ret * 10 + pop;
            }
            x = x / 10;
        }
        return ret;
    }
}